packages/teams-js/test/public/pages.spec.ts

```diff
@@ -210,42 +335,917 @@ describe('AppSDK-TeamsAPIs', () => {
+            await utils.initializeWithContext(context);
+
+            let handlerCalled = false;
+            pages.config.registerOnRemoveHandler(removeEvent => {
+              removeEvent.notifyFailure('someReason');
+              handlerCalled = true;
+            });
+
+            utils.sendMessage('settings.remove');
+
+            expect(handlerCalled).toBe(true);
+            let message = utils.findMessageByFunc('settings.remove.failure');
+            expect(message).not.toBeNull();
+            expect(message.args.length).toBe(1);
+            expect(message.args[0]).toBe('someReason');
+          });
+        } else {
+          it(`pages.config.registerOnRemoveHandler does not allow calls from ${context} context`, async () => {
+            await utils.initializeWithContext(context);
+            expect(() => pages.config.registerOnRemoveHandler(emptyCallback)).toThrowError(
+              `This call is only allowed in following contexts: ${JSON.stringify(
+                allowedContexts,
+              )}. Current context: "${context}".`,
+            );
+          });
+        }
+      });
+    });
+
+    describe('Testing pages.config.registerChangeConfigHandler function', () => {
+      const allowedContexts = [FrameContexts.content];
+
+      it('pages.config.registerChangeConfigHandler should not allow calls before initialization', () => {
+        expect(() => pages.config.registerChangeConfigHandler(emptyCallback)).toThrowError(
+          'The library has not yet been initialized',
+        );
+      });
+
+      Object.values(FrameContexts).forEach(context => {
+        if (allowedContexts.some(allowedContexts => allowedContexts === context)) {
+          it(`pages.config.registerChangeConfigHandler should successfully register a change settings handler when initialized with ${context} context`, async () => {
+            await utils.initializeWithContext(context);
+            let handlerCalled = false;
+
+            pages.config.registerChangeConfigHandler(() => {
+              handlerCalled = true;
+            });
+
+            utils.sendMessage('changeSettings', '');
+            expect(handlerCalled).toBeTruthy();
+          });
+        } else {
+          it(`pages.config.registerChangeConfigHandler does not allow calls from ${context} context`, async () => {
+            await utils.initializeWithContext(context);
+            expect(() => pages.config.registerChangeConfigHandler(emptyCallback)).toThrowError(
+              `This call is only allowed in following contexts: ${JSON.stringify(
+                allowedContexts,
+              )}. Current context: "${context}".`,
+            );
+          });
+        }
+      });
+    });
+
+    describe('Testing pages.config.isSupported function', () => {
+      it('pages.config.isSupported should return false if the runtime says pages.config is not supported', () => {
+        utils.setRuntimeConfig({ apiVersion: 1, supports: { pages: {} } });
```

Comment:
Can you please add the test case that verifies `pages.config.isSupported` returns false when `supports: {}`?

Reply:
Added in the latest commit